modules/services/agentless-scanning/main.tf

```diff
@@ -20,7 +20,7 @@
 #-----------------------------------------------------------------------------------------------------------------------
 
 data "aws_iam_policy_document" "scanning" {
-  count = (var.deploy_global_resources || var.is_organizational && var.mgt_stackset) ? 1 : 0
+  count = (var.deploy_global_resources || var.is_organizational && var.mgt_stackset && !var.delegated) ? 1 : 0
```

Comment:
nit: this may yield the correct the logic, but that's not obvious.  This could be made easier to read.

Reply:
@cgeers do you have any ideas? I thought of making a local var where if delegated is set then mgt_stackset will be turned off. But then I think mgt_stackset would always depend on delegated. There's a second scenario where you just don't want to deploy the stacksets on mgmt accounts but delegated admin is not used.

Reply:
I know at least part of this is NYC, but the different logical operators chained together (`||` with `&&`) _without_ explicit grouping makes the _intention_ less clear than it could be if there were grouping. Maybe this is normal and expected, but it strikes me as an opportunity for improvement.

Reply:
I think having a local variable based on the outcome of `var.mgt_stackset && !var.delegated_admin` like you suggested would be better especially since we want to allow deploying said stackset resources only in that scenario.

I'm not sure how deterministic is TF's parsing of multiple logical operators e.g if it takes && over || as precedence. So that's a fair good reason as well to check if this will work as expected in all combinations.

Reply:
thanks for the suggestions @cgeers @ravinadhruve10 I made the change here https://github.com/draios/terraform-aws-secure-for-cloud/pull/78/commits/82555dc2280690a7ba4a44859a81080712037a7c